File: mc/rmcp/src/main/java/calebxzhou/rdi/mc/common2/mcp/RMcpContainerTakeBatchData.java
package calebxzhou.rdi.mc.common2.mcp;

import java.util.List;

public record RMcpContainerTakeBatchData(
        String action,
        List<FailedMove> failedMoves
) {
    public record FailedMove(
            int index,
            String code
    ) {
    }
}

File: mc/rmcp/common/src/main/kotlin/calebxzhou/rdi/mc/common2/mcp/RMcpContainerTakeBatchData.kt
package calebxzhou.rdi.mc.common2.mcp

@JvmRecord
data class RMcpContainerTakeBatchData(
    val action: String,
    val failedMoves: List<FailedMove>
) {
    @JvmRecord
    data class FailedMove(
        val index: Int,
        val code: String?
    )
}
